src/io/copy.php

```diff
@@ -0,0 +1,36 @@
+<?hh
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace HH\Lib\IO;
+
+use namespace HH\Lib\Str;
+
+/**
+ * Copies content from the given $source read handle to the $target write handle
+ * until the eol of $source is reached.
+ *
+ * If `$chunk_size` is `null`, there is no limit on the chunk size - this function will
+ * copy the content of $source all at once.
+ */
+async function copy(
+  <<__AcceptDisposable>> ReadHandle $source,
+  <<__AcceptDisposable>> WriteHandle $target,
+  ?int $chunk_size = null,
```

Comment:
`->readAsync()` will change to having a default read size and `->readAllAsync()` will do the current `->readAsync()` behavior.
This would be affected by that.

Reply:
we can change them both together at the same time. can't specify a default for now since `readAsync` doesn't have one.

Reply:
There is a default, it's just intentionally hidden as it'll become a dynamic default in the future (at least where fstat() reports a block size)